### ISSUE TITLE
VRTK error

### ISSUE THREAD

**Issue by Nurul003** [thestonefox/VRTK#1847]
* Source of VRTK (Unity Asset Store) 
 * Version of VRTK (Unity Asset Store v3.2.0)
 * Version of Unity3D (Unity 20181.6f1 personal )
 * Hardware used (Oculus)
 * SDK used (SteamVR Oculus Utilities)

1.Scripts/VRTK/Scripts/Interactions/Highlighters/VRTK_MaterialColorSwapHighlighte.cs(146,66): error CS1540: Cannot access protected member `UnityEngine.Texture.Texture()' via a qualifier of type `UnityEngine.Texture'. The qualifier must be of type `VRTK.Highlighters.VRTK_MaterialColorSwapHighlighter' or derived from it

2./Scripts/VRTK/Scripts/Interactions/Highlighters/VRTK_MaterialColorSwapHighlighte.cs(146,66): error CS0122: `UnityEngine.Texture.Texture()' is inaccessible due to its protection level

I am facing above this two errors how can I solve this problem.
Thanks in Advance.
a.nurulhudahossain016.o@gmail.com

**Comment by bddckr** [comment]
Issues like these are better discussed in [![Slack](http://sysdia2.co.uk/badge.svg)](http://invite.vrtk.io) because you can instantly get answers instead of waiting on here. It's also way easier for troubleshooting in general. GitHub issues are only useful for proper bug reports with steps to reproduce in an example scene like the Issue Template requires.

tl;dr: Use the GitHub master branch to get and use the latest VRTK.

**Comment by Kishaihd** [comment]
Did anything ever come of this Error? I'm having the same error.

**Comment by bddckr** [comment]
As I already said: Use the GitHub master branch. The Asset Store version is old.

**Comment by KavanBahrami** [comment]
Its useful to see these post, even if they're a bit dated/redundant, since they pop up quickly when doing google searches on the issues (and some people new to VRTK might not know about the slack group or the versioning issues with the asset store vs git)

**Comment by imperidal** [comment]
Go to the script that you have the problem with, and replace the line with this:

`renderer.material.SetTexture("_MainTex", Texture2D.whiteTexture);`

**Comment by bddckr** [comment]
That would mean you continue to use the VRTK version that is **a year old**. Instead follow ,y recommendation to get the latest VRTK by getting it from the GitHub branch.

**Comment by tlf30** [comment]
Although it is available on GitHub, will the version on the asset store be updated anytime soon?

**Comment by Lock3113** [comment]
Some people use old versions of software to maintain a thing called "Version Control". It's foolish to assume that everyone is going to update their whole project over a simple 1 line of code error.

**Comment by bddckr** [comment]
Some people use open source projects while understanding the implicit agreement that comes with it. It's foolish to assume that every maintainer is able to maintain old versions forever or specifically work for you and your interest.

For anyone who isn't an ass about it: the mentioned bug was fixed almost a year ago and we are resource constrained with literally only two persons working on this in their free time. We do want you to join the thousands of users so you can get their help, too. They aren't active on GitHub for various reasons.

For anyone who knows what they're doing: fork and fix the old version that you need to keep using despite there being a non-breaking upgrade path to version 3.3 that is the current stable h you already know about as you read releases, release notes and get help from the community already!

**Comment by rapoulson** [comment]
Hey, I've maintained open source projects and I get that working with the community can be frustrating. Is it possible to update the notes section in the asset store? Or the Academy/Docs could get a giant banner that says "Download from Github ONLY"/"Ask questions on Slack ONLY"? 

I'm evaluating this tool for use in a university course and after downloading the asset store version (which says 3.3 released Feb 22 2019 which seems reasonably current-ish to me) I got this error message and Googled it, so this issue was pretty much the first current impression I got of VRTK and based on everything else I've read about the project...I don't think this is giving off the image you want to project.

Please understand I'm saying this with a lot of empathy and respect. I think you've built something that sounds really valuable and I'd love to be able to use it, but I also think that people are accustomed to asking questions about common problems with open source projects being out in the open and I'm pretty reticent to encourage students to use a project where they're exposed to this much snark by Googling an error message. I don't think it's reasonable to expect users to sign up for a service like Slack before they can get a project to work for the first time, but if that's the approach that makes the most sense for your community, it would be really nice to know it up front.

**Comment by reginalluna** [comment]
> Go to the script that you have the problem with, and replace the line with this:
> 
> `renderer.material.SetTexture("_MainTex", Texture2D.whiteTexture);`

I'm facing the same issue, and the line you gave didn't resolve the problem.
